bpfd-operator/pkg/helpers/helpers.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cilium/ebpf"
+	bpfdiov1alpha1 "github.com/redhat-et/bpfd/bpfd-operator/apis/v1alpha1"
+	bpfdclientset "github.com/redhat-et/bpfd/bpfd-operator/pkg/client/clientset/versioned"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	bpfdoperator "github.com/redhat-et/bpfd/bpfd-operator/controllers/bpfd-operator"
+)
+
+const (
+	DefaultMapDir = "/run/bpfd/fs/maps"
+)
+
+type ProgType string
+
+const (
+	Tc         ProgType = "TC"
+	Xdp        ProgType = "XDP"
+	TracePoint          = "TRACEPOINT"
+)
+
+var log = ctrl.Log.WithName("bpfd-helpers")
+
+// Get bpfd Kubernetes Client dynamically switches between in cluster and out of
+// cluster config setup.
+func GetClientOrDie() *bpfdclientset.Clientset {
+	config, err := rest.InClusterConfig()
```

Comment:
`IsBpfdDeployed()` has similar logic. Should getting `config` be pulled out into sub-function?

Reply:
Yep :)